### PR TITLE
added test

### PR DESCRIPTION
**Pull request by alexandra-protyanova**
Screenshot 2024-01-25 at 18 32 36

### CODE REVIEW

File: tests/formDataServer.test.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      it('should save data for valid expense on "POST /submit-expense" request', async() => {
+        const expense = {
+          date: '2024-01-25',
+          title: 'Test Expense',
+          amount: '100',
+        };
+        const response = await axios.post(`${HOST}/add-expense`, expense);
+
+        expect(response.status).toBe(200);
+        expect(fs.existsSync(dataPath)).toBe(true);
```
Comment (sergii-nosachenko) [inline tests/formDataServer.test.js:59]: You are writing to this file before every test case, so it always exists

File: tests/formDataServer.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        } catch (err) {
+          expect(err.response.data.length).toBeGreaterThan(0);
+
+          expect(
+            JSON.parse(fs.readFileSync(dataPath))
+          ).toStrictEqual({});
+        }
+      });
+
+      it('"POST /submit-expense should return JSON', async() => {
```
Comment (sergii-nosachenko) [inline tests/formDataServer.test.js:87]:
```suggestion
      it('"POST /submit-expense" should return JSON', async() => {
```

File: tests/formDataServer.test.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          title: 'Test Expense',
+          amount: '100',
+        };
+        const response = await axios.post(`${HOST}/add-expense`, expense);
+
+        expect(response.data).toStrictEqual(expense);
+        expect(response.headers['content-type']).toBe('application/json');
+      });
+
+      test('should return 404 for invalid url', async() => {
```
Comment (sergii-nosachenko) [inline tests/formDataServer.test.js:99]:
```suggestion
      it('should return 404 for invalid url', async() => {
```

File: tests/formDataServer.test.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      it('"POST /submit-expense should return JSON', async() => {
+        const expense = {
+          date: '2024-01-25',
+          title: 'Test Expense',
+          amount: '100',
+        };
+        const response = await axios.post(`${HOST}/add-expense`, expense);
+
+        expect(response.data).toStrictEqual(expense);
+        expect(response.headers['content-type']).toBe('application/json');
```
Comment (sergii-nosachenko) [inline tests/formDataServer.test.js:96]:
```suggestion
        expect(response.headers['content-type']).toBe('application/json');
        expect(response.data).toStrictEqual(expense);
```

File: tests/formDataServer.test.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      afterEach(() => {
+        server.close();
+      });
+
+      it('should return a form on "GET /" request', async() => {
+        const response = await axios.get(`${HOST}/`);
+
+        expect(response.status).toBe(200);
+        expect(response.data).toContain('<form');
+      });
```
Comment (sergii-nosachenko) [inline tests/formDataServer.test.js:48]: I think you can skip this test

**Review by sergii-nosachenko — APPROVED** [review]

File: tests/formDataServer.test.js (hunk 5 above)
Comment (alexandra-protyanova) [reply tests/formDataServer.test.js:48]: fear

File: tests/formDataServer.test.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        };
+        const response = await axios.post(`${HOST}/add-expense`, expense);
+
+        expect(response.status).toBe(200);
+
+        const savedData = JSON.parse(fs.readFileSync(dataPath));
+
+        Object.entries(expense).forEach(([key, value]) => {
+          expect(savedData[key]).toBe(value);
+        });
```
[the remark this answers is not in the document]
Comment (mgrinko) [reply tests/formDataServer.test.js:57]: Or use what you have in the last test
```js
        expect(savedData).toStrictEqual(expense);
        ```

**Review by mgrinko — APPROVED** [review]

**Pull request merged**